### PR TITLE
added favourite feature

### PR DESCRIPTION
**Pull request by akorneyc**
This PR include implementation of document favourite feature :
-changed model
-changed db schema
-added postman tests
-added unit tests

### CODE REVIEW

File: doc/dokuti.sql [hunk 1]
```diff
@@ -267,6 +267,19 @@ COMMENT ON COLUMN _documents.document_version_parts.checksum IS 'an MD5 digest t
 ALTER TABLE _documents.document_version_parts OWNER TO postgres;
 -- ddl-end --
 
+-- object: _documents.document_favourite | type: TABLE --
+-- DROP TABLE IF EXISTS _documents.document_favourite CASCADE;
+CREATE TABLE _documents.document_favourite (
```
Comment (gm7) [inline doc/dokuti.sql]: wondering if we should add a timestamp here... Would we want to sort favourites by the most recently added? Most recently accessed :thinking:

File: doc/dokuti_api.yml [hunk 2]
```diff
@@ -594,6 +603,118 @@ paths:
...
+              schema:
+                $ref: '#/components/schemas/ErrorResponse'
+        '500':
+          description: An unexpeced error occurred.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorResponse'
+  '/documents/{documentId}/unfavourite/{userId}':
+    patch:
```
Comment (gm7) [inline doc/dokuti_api.yml]: Think a `post` is fine here. Not sure that we need to do a "partial update" when we unfavourite a document.

File: doc/dokuti_api.yml [hunk 3]
```diff
@@ -594,6 +603,118 @@ paths:
             application/json:
               schema:
                 $ref: '#/components/schemas/ErrorResponse'
+  '/documents/{documentId}/favourite/{userId}':
+    patch:
```
Comment (gm7) [inline doc/dokuti_api.yml]: Think a `post` is fine here. Not sure that we need to do a "partial update" when we favourite a document with a user.

File: doc/dokuti.sql (hunk 1 above)
Comment (melissapalmer) [inline doc/dokuti.sql]: Should this table not be named `user_favourite` rather? Favourites on the UI are more a user thing than document thing.

File: doc/dokuti.sql [hunk 4]
```diff
@@ -267,6 +267,19 @@ COMMENT ON COLUMN _documents.document_version_parts.checksum IS 'an MD5 digest t
...
+	user_uuid uuid NOT NULL,
+	document_id uuid NOT NULL,
+	id bigint NOT NULL GENERATED ALWAYS AS IDENTITY ,
```
Comment (melissapalmer) [inline doc/dokuti.sql]: I wonder if there should be a unique constraint on `user_uuid` & `document_id` ?

File: doc/dokuti_api.yml [hunk 5]
```diff
@@ -50,6 +50,15 @@ paths:
             minLength: 0
             maxLength: 255
           example: Regulatory%20Document%20Name
+        - name: filterFavouriteUser
```
Comment (melissapalmer) [inline doc/dokuti_api.yml]: :thinking: Is it right that the limiting the docs search for favorites, is an editable flag. As you basically need to filter by the id of the logged in user. Are the security checks for this in place if done like this? (ie: can someone filter by the favorites of a different users?) I wonder if this attribute needs to be a flag rather filterByFavourites true/false?

File: doc/dokuti_api.yml (hunk 3 above)
Comment (melissapalmer) [reply doc/dokuti_api.yml]: Similar to question on comment above: should this API be able to add a favourite for any user or only the logged in user? ie: would it make sense to left off the {userId} parameter and change endpoint to just:  '/documents/{documentId}/favourite`

File: doc/dokuti_api.yml [hunk 6]
```diff
@@ -594,6 +603,118 @@ paths:
...
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorResponse'
+  '/documents/{documentId}/unfavourite/{userId}':
+    patch:
+      tags:
+        - Documents
+      summary: Remove a document from favourites
+      description: Remove a document from favourites by document Id and user Id.
+      operationId: UnFavouriteDocument
```
Comment (melissapalmer) [inline doc/dokuti_api.yml]: Needs to start with lowercase u .. to align with java naming conventions for function names.

File: doc/dokuti.sql (hunk 1 above)
Comment (akorneyc) [reply doc/dokuti.sql]: sure, I can add createdDateTime field, but , please confirm if we need do it now

File: doc/dokuti_api.yml (hunk 2 above)
Comment (akorneyc) [reply doc/dokuti_api.yml]: ok, will replace with "post"

File: doc/dokuti_api.yml (hunk 3 above)
Comment (akorneyc) [reply doc/dokuti_api.yml]: 1)ok, will replace with "post"
 2) for 2th question, Grant,  please confirm if need replace it?

File: doc/dokuti.sql (hunk 1 above)
Comment (akorneyc) [reply doc/dokuti.sql]: ok, let me replace name

File: doc/dokuti.sql (hunk 4 above)
Comment (akorneyc) [reply doc/dokuti.sql]: unique constraint is check on service

File: doc/dokuti_api.yml (hunk 6 above)
Comment (akorneyc) [reply doc/dokuti_api.yml]: will fix it

File: doc/dokuti_api.yml (hunk 5 above)
Comment (akorneyc) [reply doc/dokuti_api.yml]: yes, it could be replaced by Boolean and userId taken from Jwt

File: doc/dokuti_api.yml (hunk 3 above)
Comment (gm7) [reply doc/dokuti_api.yml]: Yes, the UserId should be pulled from the token (not from the path parameter). Also, a User cannot add / remove favourites from another user.

File: doc/dokuti.sql (hunk 1 above)
Comment (gm7) [reply doc/dokuti.sql]: No, not required now.

File: doc/dokuti_api.yml (hunk 5 above)
Comment (gm7) [reply doc/dokuti_api.yml]: Agree

**Pull request merged**